Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Animations;
+
+public class Checkpoint : MonoBehaviour {
+    [SerializeField]
+    public bool isActivated = false;
```

Comment:
you don't want to make public fields also serialized fields, because that's redundant. But also, I don't want public fields period. You actually have a good example here. The preferred way is to use a Property (which is just a fancy, built-into-the-language feature for creating methods like you have `SetAsActivated` below.

It's common to want to do stuff every time a value changes. This is why in C++ you make methods like you have here, and in C# you use Properties, because that's what they were made for (since this is so common).

Imagine another developer is working on your game and doesn't realize they are supposed to use `SetAsActivated`, and just go ahead and directly set `isActivated`. It's a likely scenario, since `isActivated` is exposed as part of the public API for `CheckPoint`.

Instead, we want to make `isActivated` private so this never happens. Then if you need it exposed to the editor, you mark it as a serialized field. Then you use a property (or a method like you have here) as the public API for changing isActivated, and since it's the only choice that sounds right that you can see from other classes, that's what people will use.

Reply:
Oops, I think that was before I started using TODO comments to mark what I've done to debug so I can remove it later